packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,46 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
 
 export type HeaderProps = {
+	product?: Product;
 	title: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { product, title }: HeaderProps ) {
+	const { isProductLocked, isSaving } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord } = select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore
```

Comment:
I think I needed these during testing since it did not recognize these properties, but might be good to add a quick comment to the above line for why we need to ignore.

Reply:
After rebasing this was not longer needed